---
title: C26111
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C26111
helpviewer_keywords:
- C26111
ms.assetid: 85fc740a-3bbb-41b8-a848-95e243a08da9
author: mikeblome
ms.author: mblome
manager: markl
ms.workload:
- multiple
ms.openlocfilehash: 7550fb74ae5fb6e4a18ae2d7391dfca3cc0546f3
ms.sourcegitcommit: 485ffaedb1ade71490f11cf05962add1718945cc
ms.translationtype: MT
ms.contentlocale: es-ES
ms.lasthandoff: 10/16/2019
ms.locfileid: "72448575"
---
# <a name="c26111"></a>C26111
ADVERTENCIA C26111: el autor de la llamada no puede liberar el bloqueo \<lock > antes de llamar a la función \<func >.

 La anotación `_Requires_lock_not_held_` impone una condición previa de que el recuento de bloqueos del bloqueo especificado no pueda ser mayor que cero cuando se llama a la función. Warning C26111 se emite cuando una función no puede liberar el bloqueo antes de llamar a otra función.

## <a name="example"></a>Ejemplo
 En el ejemplo siguiente se genera la advertencia C26111 porque la llamada a `DoNotLock` dentro de la sección bloqueada infringe la condición previa `_Requires_lock_not_held_`.

```

typedef struct _DATA
{
    CRITICAL_SECTION cs;
    int d;
} DATA;

_Requires_lock_not_held_(p->cs)

void DoNotLock(DATA* p)
{
    EnterCriticalSection(&p->cs);
    p->d = 0;
    LeaveCriticalSection(&p->cs);
}

void LockedFunction(DATA* p)
{
    EnterCriticalSection(&p->cs);
    DoNotLock(p); // Warning C26111
    LeaveCriticalSection(&p->cs);
}
```
